forge-game/src/main/java/forge/game/ability/AbilityUtils.java

```diff
@@ -591,6 +591,11 @@ else if (calcX[0].matches("Enchanted") || calcX[0].matches("Equipped")) {
             }
             val = handlePaid(list, calcX[1], card, ability);
         }
+        else if (calcX[0].matches("CraftedWith")) {
```

Comment:
this is no different to "ExiledWith" above

Reply:
I was getting some NPE crashes when Forge would try to calculate and getExiledCards was empty at that point... do you think it's best to just add the null checks to "ExiledWith" or keep them separate like this?

Reply:
I think as long as we reuse the implementation extra properties aren't needed
Maybe those crashes happened before you switched back?
The involved functions like `handlePaid` should all be fine with empty ones

Please post trace if it still crashes, I will debug since it shouldn't need extra guards at that place